src/Symfony/Component/ExpressionLanguage/ExpressionLanguage.php

```diff
@@ -74,7 +74,7 @@ public function evaluate($expression, $values = array())
      * Parses an expression.
      *
      * @param Expression|string $expression The expression to parse
-     * @param array             $names      An array of valid names
+     * @param array|callable    $names      An array of valid names or a callable to validate a name
```

Comment:
this is wrong, because passing a callable will actually break (``implode('-', $names)`` will not work)